CIP-0029/CIP-0029.md

```diff
@@ -0,0 +1,103 @@
+---
+CIP: 29
+Title: Phase-1 Monetary Scripts Serialization Formats
+Authors: Matthias Benkort <matthias.benkort@iohk.io>
+Comments-URI: https://github.com/cardano-foundation/CIPs/pulls/117
+Status: Draft
+Type: Standards
+Created: 2020-08-17
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This specification describes how to serialize Phase-1 monetary scripts (a.k.a. _"native scripts"_) to various formats (JSON, CBOR) to facilitate inter-operability between applications. 
+
+# Motivation
+
+While the existence of scripts is well-known, there's no agreed upon format when it comes to how those scripts shall be shared across applications. Plus, the on-chain representation of such script isn't really documented anywhere at the moment which makes it challenging for DApps to use them. 
+
+# Specification
+
+This specification covers at present two serialization formats: JSON and CBOR. The CBOR matches exactly the on-chain representation and is extracted from the cardano-ledger-specs source code and put here as a convenient place to lookup while the source can change location over time. It also covers the _full serialization_ (with serialization tag) which is missing from the CDDL specification.
+
+## CBOR
+
+The CBOR serialization format is given as a [CDDL specification in annexe](./phase-1-monetary-scripts.cddl). When a hash of the phase-1 monetary script is needed, it usually refers to a Blake2b-192 digest of the corresponding serialized script bytestring prefixed by a nul byte: `\x00`. 
+
+## JSON 
+
+The JSON format is given as a [JSON schema in annexe](./phiase-1-monetary-scripts.json). It is preferred in user interfaces such as command-lines or APIs, where some level of human inspection may be required. 
+
+## Notes
+
+- Scripts may contain unbounded integers! Implementation parsing them, either from CBOR or JSON shall be prepared to handle possible big integers (> 2^64).
+
+# Rationale
+
+- When serialized to binary, scripts are prefixed with `\x00` to distinguish them from phase-2 monetary scripts (a.k.a Plutus Script) which are then prefixed with `\x01`. This is merely a discriminator tag.
```

Comment:
This is unclear. Presumably this refers only to computing script hashes. Elsewhere it's all CBOR and covered in the CDDL.